apps/rush-lib/src/data/ApprovedPackagesPolicy.ts

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+import * as fsx from 'fs-extra';
+
+import RushConfigurationProject, { IRushConfigurationProjectJson } from './RushConfigurationProject';
+import { PackageReviewConfiguration } from './PackageReviewConfiguration';
+import { RushConstants } from '../RushConstants';
+import RushConfiguration, { IRushConfigurationJson, IApprovedPackagesPolicyJson } from './RushConfiguration';
+
+/**
+ * This is a helper object for RushConfiguration.
+ * It exposes the "approvedPackagesPolicy" feature from rush.json.
+ */
+export class ApprovedPackagesPolicy {
+  private _enabled: boolean;
+  private _ignoredNpmScopes: Set<string>;
+  private _reviewCategories: Set<string>;
+  private _browserApprovedPackages: PackageReviewConfiguration;
+  private _nonbrowserApprovedPackages: PackageReviewConfiguration;
+
+  public constructor(rushConfiguration: RushConfiguration, rushConfigurationJson: IRushConfigurationJson) {
```

Comment:
Not really -- this is just part of RushConfiguration that has been pulled out into its own class.  Nobody should be calling this constructor except for the RushConfiguration constructor.

---
In reply to: [116285163](https://github.com/Microsoft/web-build-tools/pull/181#discussion_r116285163) [](ancestors = 116285163)